### DISCUSSION TITLE
Removal of Data

### DISCUSSION THREAD

**Discussion by astupidmoose** [btcpayserver/btcpayserver#2665] [Ideas / Feature Requests]
As a merchant operates in certain jurisdictions, they have a responsibility to remove data when asked. (GDPR, CCPA etc.)

Since BTCPay stores specific data, this data also needs to be removed. 

For my purposes, a complete erasure of personal data would work, however @pavlenex and @Zaxounette both expressed concerned that this should be selective with specific data. 

Ideally, I would like to be able to search an invoice in the main invoice list and see a list of results and use the checkboxes to select all results and "remove personal data"

Putting this up here for discussion.

**Comment by Kukks** [comment]
Just for reference, the API already allows you to update the Metadata of an
invoice.

**Reply by astupidmoose** [reply]
Good to know, thanks! 

Perhaps we can create some sort of hook in the woocommerce plugin so that if a "remove personal data" action is initiated, it also tries to find the invoice in BTCPay and also changes that meta as well.

**Reply by Kukks** [reply]
Unfortunately the woocommerce plugin still needs to be rewritten to use GreenField before this can be done since it is a greenfield only feature

**Reply by bolatovumar** [reply]
@Kukks @astupidmoose given that the GreenField API can do this already it should be fairly straightforward to get this functionality going through the BTCPay UI as well, no?

**Reply by ndeet** [reply]
I will explore this once we have the Greenfield PHP library in place. 

One thought on this is if that should not also be a functionality in BTCPay Server itself? If you have some old invoices in there not connected to a live shop anymore you can't delete personal data anymore. This way all downstream plugins need to implement that functionality instead of this being available independently in BTCPay Server UI.

Another thought related to the WooCommerce plugin, would it make sense to have an option to disable sending customer data to BTCPay Server at all? Even with using the old BitPay library for the Drupal integration I added that option there so the merchant can decide to not send any customer data to BTCPay Server in the first place. I think most merchants will either handle their customer data entirely in the store OR have their accounting fetching customer data from the store. Only  a few if any will fetch customer data from BTCPay Server (not sure if there is any legal requirement to do this anyway but I could be totally wrong here).

**Reply by Kukks** [reply]
> I will explore this once we have the Greenfield PHP library in place.
> 
> One thought on this is if that should not also be a functionality in BTCPay Server itself? If you have some old invoices in there not connected to a live shop anymore you can't delete personal data anymore. This way all downstream plugins need to implement that functionality instead of this being available independently in BTCPay Server UI.
> 
> Another thought related to the WooCommerce plugin, would it make sense to have an option to disable sending customer data to BTCPay Server at all? Even with using the old BitPay library for the Drupal integration I added that option there so the merchant can decide to not send any customer data to BTCPay Server in the first place. I think most merchants will either handle their customer data entirely in the store OR have their accounting fetching customer data from the store. Only a few if any will fetch customer data from BTCPay Server (not sure if there is any legal requirement to do this anyway but I could be totally wrong here).

This is the way to do it: The order is inside an invoice should be the only requirement from integrations since these integrations store data "offchain"
